packages/babel-parser/src/parser/expression.js

```diff
@@ -807,6 +809,9 @@ export default class ExpressionParser extends LValParser {
         }
         return this.finishNode(node, "Super");
 
+      case tt.question:
+        node = this.startNode();
+        this.raise(node.start, "Partial Application syntax is not allowed");
```

Comment:
I wanted to be able to throw a message whenever the user is using an invalid syntax.
I added a boolean `allowPlaceholder` to `parseExprListItem` and `parseCallExpressionArguments` and thought, I could set it to false whenever we do not want to allow the usage of `?`.
Maybe I could throw the error message with another conditional here:

```javascript
} else if (this.match(tt.question)) {
      if (!allowPlaceholder){
        const node = this.startNode();
        this.raise(node.start, "Partial Application syntax is not allowed");
      }
      this.expectPlugin("partialApplication");
      const node = this.startNode();
      this.next();
      elt = this.finishNode(node, "ArgumentPlaceholder");
}
```

Reply:
The default error would be `unexpected ?`. it's ok either way.

Reply:
I like the `allowPlaceholder` idea from @byara , it can evolve into something more specific and make the users aware of when a partial application is allowed and when it's not, taking examples from https://github.com/tc39/proposal-partial-application#syntax we can tell the user we can point the user on the right direction instead of having him figure out why he's seeing an `unexpected ?` error.